pkg/otlp/model/translator/consumer.go

```diff
@@ -16,20 +16,51 @@ package translator
 
 import (
 	"context"
+	"encoding"
+	"fmt"
 
 	"github.com/DataDog/datadog-agent/pkg/quantile"
 )
 
 // MetricDataType is a timeseries-style metric type.
 type MetricDataType int
 
+var _ encoding.TextUnmarshaler = (*MetricDataType)(nil)
+var _ encoding.TextMarshaler = (MetricDataType)(Gauge)
+
 const (
 	// Gauge is the Datadog Gauge metric type.
 	Gauge MetricDataType = iota
 	// Count is the Datadog Count metric type.
 	Count
 )
 
+// UnmarshalText unmarshals a type into its lowercase type name.
```

Comment:
```suggestion
// UnmarshalText implements encoding.TextUnmarshaler.
```
This is what I usually see when documenting the implementation of an interface. Then, the interface itself has the documentation for the method. But I don't feel strongly about this.